x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/ClientTransformIndexer.java

```diff
@@ -111,6 +113,19 @@
 
         // TODO: move into context constructor
         context.setShouldStopAtCheckpoint(shouldStopAtCheckpoint);
+
+        if (transformConfig.getSettings().getUsePit() != null) {
+            disablePit = transformConfig.getSettings().getUsePit() == false;
+        }
```

Comment:
`getUsePit()` should know about the default value and not return a boxed boolean.

Reply:
/CC @przemekwitek WDYT? 

I think with requiring a default, the issue with `align_checkpoints` would not have happened, right?

Reply:
`boolean getUsePit(boolean default)`

If we are going to keep them boxed, Java already has a way to handle this with `Optional`. 

So

```
Optional<Boolean> getUsePit()
```
```
getUsePit().orElse(true)
```
For epoch millis
```
getDatesAsEpochMillis().orElse(version.before(Version.V_7_11_0))
```